components/cam/src/physics/cam/convect_deep.F90

```diff
@@ -93,8 +100,25 @@ subroutine convect_deep_register
   select case ( deep_scheme )
   case('ZM') !    Zhang-McFarlane (default)
      call zm_conv_register
-  end select
+  case('off')
+     call pbuf_add_field('DP_CLDLIQ','global',dtype_r8,(/pcols,pver/), dp_cldliq_idx)    
+     call pbuf_add_field('DP_CLDICE','global',dtype_r8,(/pcols,pver/), dp_cldice_idx)
+     call pbuf_add_field('DP_FLXPRC','global',dtype_r8,(/pcols,pver/), dp_flxprc_idx)
+     call pbuf_add_field('DP_FLXSNW','global',dtype_r8,(/pcols,pver/), dp_flxsnw_idx)
+  end select 
+
+! Flux of precipitation from deep convection (kg/m2/s)
+   call pbuf_add_field('DP_FLXPRC','global',dtype_r8,(/pcols,pverp/),dp_flxprc_idx) 
```

Comment:
Am I missing something here? It looks like you are adding these fields to pbuf twice, which should raise a runtime error. Also, two questions: 1) why not just call zm_conv_register, which adds these fields to pbuf anyways, and 2) can we do away with the dependence on deep convective fields existing in pbuf when not using deep convection? For COSP I think it will be pretty straightforward, but I'm not sure how much further ZM reaches its tentacles.

Reply:
good gravy, thanks for noting this.  I removed the pbuf_add_field.  These fields were being added twice but no run time error was being produced (I looked into pbuf code and didn't see a runtime error for duplicate adds).  1) My logic here was that PBUF fields that are specific to ZM (i.e. only used in ZM interface and ZM code) would be added in zm_conv_register.  PBUF fields related to deep convection but are expected in other parts of the E3SM code would then be added in the general convect_deep_register.  I added a comment stating this logic. 2) This would be a fairly moderate to large effort to do and test seeing the number of routines these variables touch.  I personally don't see this as a good use of SCREAM time and don't really see any particular benefit to SCREAM since we will eventually abandon the F90 code.

Reply:
Ah yeah that makes sense, thanks for clarifying. Changes look good.